qualang_tools/octave_tools/calibration_result_plotter.py

```diff
@@ -0,0 +1,504 @@
+        ]
+    )
+
+    dp = np.mean(np.diff(r.p_scan, axis=1))
+    dg = np.mean(np.diff(r.g_scan, axis=0))
+
+    plt.xlabel("phase (rad)")
+    plt.ylabel("gain(%)")
+    plt.axis("equal")
+
+    plt.contour(r.p_scan, r.g_scan, r.image * 1e6, colors="w", alpha=0.3)
+    plt.pcolor(r.p_scan, r.g_scan, r.image * 1e6)
+
+    plt.plot(r.phase, r.gain, "yo", markersize=8)
+    plt.plot(r.phase, r.gain, "ro", markersize=4)
+
+    plt.text(
+        r.phase,
+        r.gain - 2 * dg,
+        f"phase={r.phase:.5f}\ngain={r.gain:.5f}",
+        color="y",
+        horizontalalignment="center",
+        verticalalignment="top",
+    )
+
+    plt.colorbar()
+
+    plt.text(
+        np.min(r.p_scan) + 1.5 * dp,
+        np.max(r.g_scan - 1.5 * dg),
+        f"{label}fine\nLO = {lo_freq/1e9:.3f}GHz\nIF = {if_freq/1e6:.3f}MHz",
+        color="k",
+        bbox=dict(facecolor="w", alpha=0.2),
+        verticalalignment="top",
+    )
+
+    if if_freq_data.prev_result is not None:
+
+        p_gain = if_freq_data.prev_result[0]
+        p_phase = if_freq_data.prev_result[1]
+
+        plt.plot(
+            p_phase + cross[0] / 5 * dp * 2,
+            p_gain + cross[1] / 5 * dg * 2,
+            "--",
+            color="#aaaaaa",
+            linewidth=0.5,
+        )
+
+    plt.subplot(223)
+
+    X, Y = r.p_scan, r.g_scan
+    p = r.fit.pol
+
+    image = p[0] + p[1] * X + p[2] * Y + p[3] * X**2 + p[4] * X * Y + p[5] * Y**2 - r.image
+
+    plt.pcolor(r.p_scan, r.g_scan, image * 1e6)
+    plt.xlabel("phase (rad)")
+    plt.ylabel("gain")
+    plt.axis("equal")
+
+    plt.plot(r.phase, r.gain, "yo", markersize=8)
+    plt.plot(r.phase, r.gain, "ro", markersize=4)
+
+    plt.colorbar()
+
+    plt.text(
+        np.min(r.p_scan) + 1.5 * dp,
+        np.max(r.g_scan - 1.5 * dg),
+        f"{label}fit error\nLO = {lo_freq/1e9:.3f}GHz\nIF = {if_freq/1e6:.3f}MHz",
+        color="k",
+        bbox=dict(facecolor="w", alpha=0.2),
+        verticalalignment="top",
+    )
+
+    plt.subplot(221)
+
+    plt.xlim(0, 1)
+    plt.ylim(0, 1)
+
+    content = [
+        "IMAGE auto calibration",
+        f"      LO = {lo_freq/1e9:.3f}GHz, IF = {if_freq/1e6:.3f}MHz",
+        "",
+        "Current result:",
+        f"      gain = {r.gain*100:.02f}%, phase = {r.phase*180.0/np.pi:.2f}deg",
+        "Previous result:",
+    ]
+
+    if if_freq_data.prev_result is not None:
+
+        p_gain = if_freq_data.prev_result[0]
+        p_phase = if_freq_data.prev_result[1]
+
+        content.append(f"      gain = {p_gain*100:.02f}%, phase = {p_phase*180.0/np.pi:.2f}deg")
+        content.append("Diff:")
+        content.append(f"      gain = {(p_gain-r.gain)*100:.02f}%, phase = {(p_phase-r.phase)*180.0/np.pi:.2f}deg")
+
+    else:
+        content.append("      - none -")
+
+    plt.text(
+        0.05,
+        0.75,
+        "\n".join(content),
+        horizontalalignment="left",
+        verticalalignment="top",
+    )
```

Comment:
Maybe it will look nicer with a box: `bbox=dict(facecolor='none', edgecolor='black', boxstyle='round,pad=1')`